descartes_planner/src/sparse_planner.cpp

```diff
@@ -30,18 +30,39 @@ using namespace descartes_trajectory;
 
 namespace
 {
-    std::vector<descartes_core::TimingConstraint>
-    extractTiming(const std::vector<descartes_core::TrajectoryPtPtr>& points)
+  descartes_core::TimingConstraint
+  cumulativeTimingBetween(const std::vector<descartes_core::TrajectoryPtPtr>& dense_points,
+                          size_t start_index,
+                          size_t end_index)
+  {
+    descartes_core::TimingConstraint tm (0.0, 0.0);
+    for (size_t i = start_index + 1; i <= end_index; ++i)
     {
-      std::vector<descartes_core::TimingConstraint> result;
-      result.reserve(points.size());
-
-      for (const auto& ptr : points)
+      const descartes_core::TimingConstraint& pt_tm = dense_points[i]->getTiming();
+      if (pt_tm.isSpecified())
       {
-        result.push_back(ptr->getTiming());
+        // Add time as normal
+        tm.upper += pt_tm.upper;
+        tm.lower += pt_tm.lower;
+      }
+      else
+      {
+        // A single unspecified timing makes the range unspecified
+        tm = descartes_core::TimingConstraint();
+        break;
       }
-      return result;
     }
+    return tm;
+  }
+
+  descartes_core::TrajectoryPtPtr
+  copyAndSetTiming(const descartes_core::TrajectoryPtPtr& ptr, const descartes_core::TimingConstraint& tm)
```

Comment:
While I hesitate to add too many additional methods to trajectory pt, I think this one might be common enough to be worth it.

I toyed with the notion of having a new copy() function that takes a timing value as an arg:

```c++
auto copy = original_pt->copy(TimingContraint(1.0));
```

While I like the brevity of this, I think it's better to be explicit. +1